string type
string[] args
bool is_negative